File: sequence/token.php
<?php

/*
Token
Autor: Sérgio Mandrake <sergio_mandrake@yahoo.com.br>
Tipo: sequence
Descrição: Gera um token aleatório
Complexidade:
    Pior caso: ?
    Melhor caso: ?
Dificuldade: fácil
*/

function token($comprimento = 6) {
    $caracteres = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
    $totalCarac = strlen($caracteres) -1;
    $resultado = '';
    for ($i = 0; $i < $comprimento; $i++) {
        $resultado .= $caracteres[mt_rand(0, $totalCarac)];
    }

    return $resultado;
}

echo token(5);

File: algorithms/sequence/token.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

"""
Token

Autor:
    ?
Colaborador:
    Felipe Djinn <felipe@felipedjinn.com.br>
Tipo:
    sequence
Descrição:
    Gera um token aleatório
Complexidade:
    ?
Dificuldade:
    facil
"""

import random
import string

def token(length = 10):
 return ''.join(random.choice(string.letters) for i in xrange(length)) 


"""
Examples
"""

print "Token com 10 caracteres (padrão): " + token()
print "Token com 5 caracteres: " +token(5)
print "Token com 15 caracteres " +token(15)
